server/routes/wishCards.js

```diff
@@ -80,89 +102,102 @@ const AgencyRepository = require('../db/repository/AgencyRepository');
 
 // @desc    wishcard creation form sends data to db
 // @route   POST '/wishcards'
-// @access  Private, must be verified as a partner
+// @access  Private, verified partner
 // @tested 	Yes
-router.post('/', upload.single('wishCardImage'), async (req, res) => {
-  if (req.file === undefined) {
-    handleError(
-      res,
-      400,
-      `Error: File must be in jpeg, jpg, gif, or png format. The file
+router.post(
+  '/',
+  checkPermissions,
+  upload.single('wishCardImage'),
+  createWishcardValidationRules(),
+  validate,
+  async (req, res) => {
+    if (req.file === undefined) {
+      handleError(res, 400, {
+        success: false,
+        error: `Error: File must be in jpeg, jpg, gif, or png format. The file
         must also be less than 5 megabytes.`,
-    );
-  } else {
-    try {
-      const newWishCard = await WishCardRepository.createNewWishCard({
-        childBirthday: new Date(req.body.childBirthday),
-        wishItemPrice: Number(req.body.wishItemPrice),
-        wishCardImage: req.file.location,
-        createdBy: res.locals.user._id,
-        // Uncomment once address fields are added to profile page.
-        /* address: {
+      });
+    } else {
+      try {
+        let newWishCard = await WishCardRepository.createNewWishCard({
```

Comment:
why let if the variable is never changed again?

Reply:
Changed it to const :+1: